### PR TITLE
Feature/error-status

### PR DESCRIPTION
**Pull request by jdnavarro**
Now the `Sequence.status` and `Step.status` methods return the `Status`es from the `SharedModel` including the error status.

`_.status` wasn't used anywhere in the engine so it was easy to switch it directly to the `Status`es from the `SharedModel`. Internally the engine keeps using the keeps using its previous `Status` structure.

### CODE REVIEW

File: modules/edu.gemini.seqexec.engine/src/main/scala/edu/gemini/seqexec/engine/Step.scala [hunk 1]
```diff
@@ -14,10 +16,18 @@ object Step {
...
     *
     */
-  def status(step: Step[Action \/ Result]): Status =
-    if (step.executions.isEmpty || step.all(_.isLeft)) Status.Waiting
-    else if (step.all(_.isRight)) Status.Completed
-    else Status.Running
+  def status(step: Step[Action \/ Result]): StepState = {
+
+    // At least an Action in this Step errored.
+    if (step.executions.isEmpty || step.any(Execution.errored)) StepState.Error("An action errored")
```
Comment (cquiroz) [inline modules/edu.gemini.seqexec.engine/src/main/scala/edu/gemini/seqexec/engine/Step.scala]: Is it really an error that `executions` is empty?

**Comment by cquiroz** [comment]
I think it would be good to have tests showing the state when there are errors

File: modules/edu.gemini.seqexec.engine/src/main/scala/edu/gemini/seqexec/engine/Step.scala (hunk 1 above)
Comment (cquiroz) [inline modules/edu.gemini.seqexec.engine/src/main/scala/edu/gemini/seqexec/engine/Step.scala]: 👍

**Comment by jdnavarro** [comment]
I still have to add the tests for the Sequence Status before merging.

**Pull request merged**